### PR TITLE
fix: set excess blob gas and blob gas used for empty payloads

### LINKED ISSUE #6324: Missing cancun header fields if empty payload
### Describe the bug

when building empty payload blob fields are set to None

https://github.com/paradigmxyz/reth/blob/9f82af20f96919146d82cb2cdfc00bc952264c5e/crates/payload/ethereum/src/lib.rs#L141-L142

resulting in empty list code instead of 0 value in encoding resulting in bad hash

https://github.com/paradigmxyz/reth/blob/9f82af20f96919146d82cb2cdfc00bc952264c5e/crates/primitives/src/header.rs#L462-L462

### Steps to reproduce

-

### Node logs

```text
-
```

### Platform(s)

Linux (x86)

### What version/commit are you on?

- 

### What database version are you on?

- 

### What type of node are you running?

Archive (default)

### What prune config do you use, if any?

- 

### If you've built Reth from source, provide the full command you used

-

### Code of Conduct

- [X] I agree to follow the Code of Conduct

### PR DESCRIPTION
**Pull request by Rjected**
Fixes https://github.com/paradigmxyz/reth/issues/6324

### CODE REVIEW

**Comment by Rjected** [comment]
re failing test, this is why we can't encode `0` if None:
then, when decoding, `Some(0)` and `None` are indistinguishable

File: crates/primitives/src/header.rs [hunk 1]
```diff
@@ -423,6 +423,12 @@ impl Encodable for Header {
         self.mix_hash.encode(out); // Encode mix hash.
         B64::new(self.nonce.to_be_bytes()).encode(out); // Encode nonce.
 
+        // The following code is needed only to handle proptest-generated headers that are
+        // technically invalid.
+        //
+        // TODO: make proptest generate more valid headers, ie if there is no base fee, there
+        // should be no withdrawals root or any future fork field.
```
Comment (mattsse) [inline crates/primitives/src/header.rs:430]: can we open an issue for manual arbitrary impl?

**Review by mattsse — APPROVED** [review]
too many rules

File: crates/primitives/src/header.rs (hunk 1 above)
Comment (Rjected) [reply crates/primitives/src/header.rs:430]: yes https://github.com/paradigmxyz/reth/issues/6329

**Pull request merged**